Clarify counter

chrysn opened the pull request:
The mentions of counting Echo values was seen as confusing in the reviews, especially given that we didn't 

Open steps:

1. [ ] Add distinction (in seccons? where the features are introduced?) of what one *may* want from an Echo, and what properties are *required* to be provided ("Want aliveness? Must be unpredictable even when having seen value as another client.") (CA)
2. [x] read through it again for nits and (CA)
3. [x] check whether this is the outcome we want of yesterday's call (GS, JPM)

Commit:
Point out that a MUST on integrity protection is not a new requirement of seccons

Commit:
Add counter Echo construction example

gselander committed:
fixing ref

gselander committed:
updated normative text on Echo values

gselander committed:
aligned terminology

gselander committed:
rephrased 'except for counter'

gselander committed:
rephrased counter-based example

gselander committed:
addressing Barry's comment

gselander commented:
I added some more commits:

* Barry’s two comments leftover from Christian’s latest fix.
* Some rephrasing in new example of persistent counters
* Rephrased occurrences of  “unless counter … “ in security considerations since it wasn’t clear if that was the only exception
* Aligned terminology

I think only the first "Open step" above remains to do.

Commit:
counting example: slightly generalizel rationalize reference to B.1.1

* Recovering volatile state loss is just one (but an important)
  application of persistent counters, and thus moved into parentheses as
  an example.

* The "Another example of" introduction didn't capture why this is
  referenced here -- it is for the benefit of implementors and of
  readers who might be tempted to dismiss such counters due to their
  locking difficulties.

chrysn commented:
Yes, and that's probably excessive for now.

Thanks for the updates, I'm merging this and will finish the changelog to submit as -11
